lib/database/exceptions.rb

```diff
@@ -85,7 +85,9 @@ class NonUniqueObjectException < RavenException
   end
   class ConcurrencyException < RavenException 
   end
-  class ArgumentNullException < RavenException 
+  class InvalidArgumentException < RavenException
+  end
+  class ArgumentNullException < RavenException
   end
   class ArgumentOutOfRangeException < RavenException 
```

Comment:
Most of those exceptions are mapped 1 to 1 from .NET ones, which is probably not what we want and not what Ruby developer using this lib expects. General idea for exception handling is:
- if it's server-side error that's rethrown on the client-side then it should be thrown as something inheriting `RavenException`
- if it's client-side error then we should use analogous Ruby language exceptions e.g.:
  - InvalidOperationException -> RuntimeError (I think)
  - ArgumentNullException, InvalidArgumentException -> ArgumentError
  - ArgumentOutOfRangeException -> IndexError (?)

Apart from that, do we have similar constructs in Node.js client? If so, we should take care of that as well. However it's another story, since there's only Error class there for this kind of stuff, so custom ones must be created anyway.